Source/engine/point.hpp

```diff
@@ -134,61 +118,107 @@ struct Point {
 	 * @param other Point to which we want the distance
 	 * @return Exact magnitude of vector this -> other
 	 */
-	int ExactDistance(Point other) const
+	template <typename PointCoordT>
+	int ExactDistance(PointOf<PointCoordT> other) const
 	{
 		auto vector = *this - other; // No need to call abs() as we square the values anyway
 
 		// Casting multiplication operands to a wide type to address overflow warnings
 		return static_cast<int>(std::sqrt(static_cast<int64_t>(vector.deltaX) * vector.deltaX + static_cast<int64_t>(vector.deltaY) * vector.deltaY));
 	}
 
-	constexpr friend Point abs(Point a)
-	{
-		return { abs(a.x), abs(a.y) };
-	}
-
-	constexpr int ManhattanDistance(Point other) const
+	template <typename PointCoordT>
+	constexpr int ManhattanDistance(PointOf<PointCoordT> other) const
 	{
-		Displacement offset = abs(*this - other);
+		Displacement offset { abs(*this - other) };
 
 		return offset.deltaX + offset.deltaY;
 	}
 
-	constexpr int WalkingDistance(Point other) const
+	template <typename PointCoordT>
+	constexpr int WalkingDistance(PointOf<PointCoordT> other) const
 	{
-		Displacement offset = abs(*this - other);
+		Displacement offset { abs(*this - other) };
 
 		return std::max<int>(offset.deltaX, offset.deltaY);
 	}
 
 	/**
 	 * @brief Converts a coordinate in megatiles to the northmost of the 4 corresponding world tiles
 	 */
-	constexpr Point megaToWorld() const
+	constexpr PointOf<CoordT> megaToWorld() const

```

Comment:
One of the things I was hoping we could do in the future is make these functions part of the WorldTilePosition (and maybe a future MegaTilePosition?) specialisations instead of the base template. However not sure if that would be considered desirable given multi-use variables like quest positions storing both values at different stages of dungeon generation/quest progression.

Idle thought for the future anyway.

Reply:
Sounds good, I was thinking the same thing, though rather than specializations a `struct WorldTilePosition : public Point<uint8_t>`